### PR TITLE
Read the cluster domain from the Eventing CR or from the ConfigMap

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Read the cluster domain from the Eventing CR or from the ConfigMap.

**Related issue(s)**

- https://github.com/kyma-project/eventing-manager/issues/137
- https://github.com/kyma-project/eventing-manager/issues/125

### CODE REVIEW

File: internal/controller/eventing/domain_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+	k8smocks "github.com/kyma-project/eventing-manager/pkg/k8s/mocks"
+	"github.com/kyma-project/eventing-manager/test/utils"
+)
+
+func Test_readDomainFromConfigMap(t *testing.T) {
+	// given
+	var (
+		ctx = context.TODO()
+	)
```
Comment (friedrichwilken) [inline internal/controller/eventing/domain_test.go:20]: why not just 
```suggestion
		ctx := context.TODO()
	
```

File: internal/controller/eventing/eventmesh.go [hunk 2]
```diff
@@ -59,13 +57,31 @@ func (r *Reconciler) reconcileEventMeshSubManager(ctx context.Context, eventing
...
+		r.namedLogger().Infof(
+			`Domain is not configured in the Eventing CR, reading it from the ConfigMap %s/%s`,
+			shootInfoConfigMapNamespace, shootInfoConfigMapName,
+		)
+		domain, err = r.readDomainFromConfigMap(ctx)
+		if err != nil || utils.IsEmpty(domain) {
+			return domainMissingError(err)
+		}
+	}
+	r.namedLogger().Infof(`Domain is %s`, domain)
```
Comment (friedrichwilken) [inline internal/controller/eventing/eventmesh.go:73]: Mybe we should also log fact that we got our domain from `eventing.Spec.Backend.Config.Domain` explicitly. While this is implicit, without seeing this code we would just see in in the logs what our domain is, without knowing, that it comes from `eventing.Spec.Backend.Config.Domain`.

Comment (marcobebway) [reply internal/controller/eventing/eventmesh.go:73]: IMO these logs are not really needed, I just put them there for the first releases to help us see what the domain value is, then we should reduce them to debug level.

But for the time being, we will get one of the two following logs:

When the domain is configured in the Eventing CR, we get:

```
Domain is ...
```

Or, when it is not configured in the Eventing CR but it is configured in the ConfigMap, we get:

```
Domain is not configured in the Eventing CR ...
Domain is ...
```

What is more important is the condition we get from the Eventing CR:

```
  - lastTransitionTime: "2023-10-17T15:05:06Z"
    message: domain configuration is missing. domain must be configured in either the Eventing 
    CustomResource under "Spec.Backend.Config.Domain" or in the ConfigMap 
    "kube-system/shoot-info" under "data.domain"
    reason: EventMeshSubscriptionManagerFailed
    status: "False"
    type: SubscriptionManagerReady
```

which tells the user where to put the domain value.

Comment (friedrichwilken) [reply internal/controller/eventing/eventmesh.go:73]: thanks for the detailed explanation. This is fine now :)

**Review by friedrichwilken — APPROVED** [review]

**Pull request merged**